src/Mimi.jl

```diff
@@ -216,14 +216,15 @@ function setparameter(m::Model, component::Symbol, name::Symbol, value)
 end
 
 function checklabels(m::Model, component::Symbol, name::Symbol, parametername::Symbol, p::ArrayModelParameter)
-    if !(eltype(p.values) <: getmetainfo(m, component).parameters[parametername].datatype)
+    meta_comp_name = Symbol(string(m.components2[component].component_type))
+    if !(eltype(p.values) <: getmetainfo(m, meta_comp_name).parameters[parametername].datatype)
```

Comment:
The problem seems to be that the name "adder" evaluates to Mimi.adder. So when we add the component to the model, what it tries to add is Mimi.adder, which isn't quite what we want. But I don't fully understand why the name "adder" isn't associated with the ComponentState that was created by the `defcomp adder`, like any other component normally is.